muxsearch/main.c

```diff
@@ -249,13 +251,13 @@ void reset_osk() {
 void list_nav_prev(int steps) {
     play_sound("navigate", nav_sound, 0, 0);
     for (int step = 0; step < steps; ++step) {
-        current_item_index = (current_item_index == 0) ? ui_count - 1 : current_item_index - 1;
         nav_prev(ui_group, 1);
         nav_prev(ui_group_value, 1);
         nav_prev(ui_group_glyph, 1);
         nav_prev(ui_group_panel, 1);
     }
-    update_scroll_position(theme.MUX.ITEM.COUNT, theme.MUX.ITEM.PANEL, ui_count, current_item_index, ui_pnlContent);
+    current_item_index = (int)(intptr_t)lv_obj_get_user_data(lv_group_get_focused(ui_group_panel));
+    scroll_object_to_middle(ui_pnlContent, lv_group_get_focused(ui_group_panel));
```

Comment:
updating scrolling to be based on scrolling the current item into view based on it's parent container so that it no longer depends on current_item_index